ir/node.h

```diff
@@ -110,6 +110,7 @@ class Node : public virtual INode {
     void sourceInfoToJSON(JSONGenerator &json) const;
     Util::JsonObject* sourceInfoJsonObj() const;
     virtual bool operator==(const Node &a) const { return typeid(*this) == typeid(a); }
+    virtual bool equiv(const Node &a) const { return typeid(*this) == typeid(a); }
```

Comment:
Can you add a comment, either here or wherever you think is appropriate, documenting `equiv`, including
 - the difference between it and `operator==`, and
 - the relationship between `equiv` and the visitors?

For example, it looks like `equiv` checks the equality even for fields that aren't visited.

Reply:
I think we should call this `syntacticallyEquivalent` to make it clear that this is not meant to say that the two trees evaluate to the same value or some other notion of equivalence.